From ca11871f7e19d26db9833b5b76322a1fcc3e3448 Mon Sep 17 00:00:00 2001
From: moo <moo@moo.moo>
Date: Fri, 16 Oct 2020 17:30:30 -0400
Subject: [PATCH] Update-Config


diff --git a/src/main/java/com/destroystokyo/paper/ccSpigotConfig.java b/src/main/java/com/destroystokyo/paper/ccSpigotConfig.java
index 6136334a3..9f67b151b 100644
--- a/src/main/java/com/destroystokyo/paper/ccSpigotConfig.java
+++ b/src/main/java/com/destroystokyo/paper/ccSpigotConfig.java
@@ -185,9 +185,9 @@ public class ccSpigotConfig {
         DisableChunkRelocation = getBoolean("experimental.disable-chunk-relocation", DisableChunkRelocation);
     }
 
-    public static boolean LimitEntitySpeed = true;
+    public static boolean LimitEntitySpeed = false;
     private static void EntitySpeedLimit() {
-        LimitEntitySpeed = getBoolean("settings.limit-entity-speed", LimitEntitySpeed);
+        LimitEntitySpeed = getBoolean("experimental.limit-entity-speed", LimitEntitySpeed);
     }
 
     public static boolean DebugEntitySpeed = true;
-- 
2.22.0.windows.1

